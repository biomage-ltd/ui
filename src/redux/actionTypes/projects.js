const PROJECTS = 'projects';

/**
 * Save project.
 */
const PROJECTS_CREATE = `${PROJECTS}/create`;

/**
 * Update project.
 */
const PROJECTS_UPDATE = `${PROJECTS}/update`;

/**
 * Set active project.
 */
const PROJECTS_SET_ACTIVE = `${PROJECTS}/setActive`;

/**
 * Delete project.
 */
const PROJECTS_DELETE = `${PROJECTS}/delete`;

/**
 * Add project metadata.
 */
const PROJECTS_METADATA_CREATE = `${PROJECTS}/metadata_create`;

/**
 * Update project metadata.
 */
const PROJECTS_METADATA_UPDATE = `${PROJECTS}/metadata_update`;

/**
 * Delete project metadta.
 */
const PROJECTS_METADATA_DELETE = `${PROJECTS}/metadata_delete`;

export {
  PROJECTS_CREATE,
  PROJECTS_UPDATE,
  PROJECTS_SET_ACTIVE,
  PROJECTS_DELETE,
  PROJECTS_METADATA_CREATE,
  PROJECTS_METADATA_UPDATE,
  PROJECTS_METADATA_DELETE,
};
