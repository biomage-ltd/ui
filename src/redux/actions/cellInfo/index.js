import setCellInfoFocus from './setCellInfoFocus';
import updateCellInfo from './updateCellInfo';

export {
  setCellInfoFocus,
  updateCellInfo,
};
