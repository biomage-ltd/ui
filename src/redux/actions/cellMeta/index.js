import loadCellMeta from './loadCellMeta';

export default loadCellMeta;
