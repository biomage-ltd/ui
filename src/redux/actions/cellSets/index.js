import createCellSet from './createCellSet';
import deleteCellSet from './deleteCellSet';
import resetCellSets from './resetCellSets';

import loadCellSets from './loadCellSets';
import saveCellSets from './saveCellSets';
import updateCellSetHierarchy from './updateCellSetHierarchy';
import updateCellSetProperty from './updateCellSetProperty';
import updateCellSetSelected from './updateCellSetSelected';
import setCellSetHiddenStatus from './setCellSetHiddenStatus';
import unhideAllCellSets from './unhideAllCellSets';

export {
  createCellSet,
  deleteCellSet,
  resetCellSets,
  loadCellSets,
  saveCellSets,
  updateCellSetHierarchy,
  updateCellSetSelected,
  updateCellSetProperty,
  setCellSetHiddenStatus,
  unhideAllCellSets,
};
