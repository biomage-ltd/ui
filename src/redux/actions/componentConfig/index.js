import loadPlotConfig from './loadPlotConfig';
import loadComponentConfig from './loadComponentConfig';
import updatePlotConfig from './updatePlotConfig';
import savePlotConfig from './savePlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
};
