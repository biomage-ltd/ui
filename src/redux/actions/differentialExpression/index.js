import loadDifferentialGeneExpression from './loadDifferentialExpression';
import setComparisonType from './setComparisonType';
import setComparisonGroup from './setComparisonGroup';

export {
  loadDifferentialGeneExpression,
  setComparisonType,
  setComparisonGroup,
};
