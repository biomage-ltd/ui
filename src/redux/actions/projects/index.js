import createProject from './createProject';
import updateProject from './updateProject';
import deleteProject from './deleteProject';
import saveProject from './saveProject';
import setActiveProject from './setActiveProject';
import createMetadataTrack from './createMetadataTrack';
import updateMetadataTrack from './updateMetadataTrack';
import deleteMetadataTrack from './deleteMetadataTrack';

export {
  createProject,
  updateProject,
  deleteProject,
  saveProject,
  setActiveProject,
  createMetadataTrack,
  updateMetadataTrack,
  deleteMetadataTrack,
};
