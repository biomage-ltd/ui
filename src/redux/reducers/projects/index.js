import {
  PROJECTS_CREATE,
  PROJECTS_UPDATE,
  PROJECTS_SET_ACTIVE,
  PROJECTS_DELETE,
} from '../../actionTypes/projects';
import initialState from './initialState';
import projectsCreate from './projectsCreate';
import projectsUpdate from './projectsUpdate';
import projectsDelete from './projectsDelete';
import projectsSetActive from './projectsSetActive';

const projectsReducer = (state = initialState, action) => {
  switch (action.type) {
    case PROJECTS_CREATE: {
      return projectsCreate(state, action);
    }

    case PROJECTS_UPDATE: {
      return projectsUpdate(state, action);
    }

    case PROJECTS_SET_ACTIVE: {
      return projectsSetActive(state, action);
    }

    case PROJECTS_DELETE: {
      return projectsDelete(state, action);
    }

    default: {
      return state;
    }
  }
};

export default projectsReducer;
