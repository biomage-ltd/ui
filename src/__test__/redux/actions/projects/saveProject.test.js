import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import initialProjectState from '../../../../redux/reducers/projects/initialState';
import { saveProject } from '../../../../redux/actions/projects';

jest.mock('localforage');

enableFetchMocks();
const mockStore = configureStore([thunk]);

describe('saveProject action', () => {
  const mockProject = {
    name: 'mockProject',
    description: 'Project',
    uuid: 'project-uuid',
    createdDate: '2021-01-01T00:00:00.000Z',
    lastModified: null,
    samples: [],
    lastAnalyzed: null,
  };

  const initialState = {
    projects: {
      ...initialProjectState,
      ids: [mockProject.uuid],
      meta: {
        activeprojectUuid: null,
      },
      [mockProject.uuid]: mockProject,
    },
  };

  beforeEach(() => {
    const response = new Response(JSON.stringify({ one: 'one' }));

    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValueOnce(response);
  });

  it('Dispatches fetch correctly.', async () => {
    const store = mockStore(initialState);
    await store.dispatch(saveProject(mockProject.uuid));

    expect(fetchMock).toHaveBeenCalledWith(
      `http://localhost:3000/v1/projects/${mockProject.uuid}`,
      {
        body: JSON.stringify(mockProject),
        headers: {
          'Content-Type': 'application/json',
        },
        method: 'PUT',
      },
    );
  });

  it('Dispatches a notification when fetch fails.', async () => {
    fetchMock.resetMocks();
    fetchMock.mockReject(new Error('some weird error that happened'));

    const store = mockStore(initialState);
    await store.dispatch(saveProject(mockProject.uuid));

    const firstAction = store.getActions()[0];
    expect(firstAction).toMatchSnapshot();
  });
});
