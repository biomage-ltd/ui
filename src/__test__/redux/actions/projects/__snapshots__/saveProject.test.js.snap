// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saveProject action Dispatches a notification when fetch fails. 1`] = `
Object {
  "payload": Object {
    "message": "Could not connect to the server. Check your internet connection.",
    "time": 5,
    "type": "error",
  },
  "type": "notifications/pushMessage",
}
`;
